### PR TITLE
iproute2: update to 6.9.0

### PR DESCRIPTION
**Pull request by rsalvaterra**
Support for iptables action has been dropped. Remove tc-mod-iptables and related
patch (175-reduce-dynamic-syms.patch).

Include a patch to fix the build with GCC 14.

Fix 170-ip_tiny.patch, as the help text didn't match all the included functions.

All other patches automatically rebased.

Signed-off-by: Rui Salvaterra \<rsalvaterra@gmail.com\>

### CODE REVIEW

**Comment by hauke** [comment]
This does not build:
```
make[5]: Leaving directory '/__w/openwrt/openwrt/openwrt/build_dir/target-mips-openwrt-linux-musl_musl/linux-malta_be/iproute2-tcfull/iproute2-6.8.0/rdma'
make[4]: Leaving directory '/__w/openwrt/openwrt/openwrt/build_dir/target-mips-openwrt-linux-musl_musl/linux-malta_be/iproute2-tcfull/iproute2-6.8.0'
cp: cannot stat '/__w/openwrt/openwrt/openwrt/build_dir/target-mips-openwrt-linux-musl_musl/linux-malta_be/iproute2-tcfull/iproute2-6.8.0/tc/m_xt.so': No such file or directory
make[3]: *** [Makefile:277: /__w/openwrt/openwrt/openwrt/bin/packages/mips_24kc/base/tc-mod-iptables_6.8.0-r1_mips_24kc.ipk] Error 1
time: package/network/utils/iproute2/tcfull/compile#30.12#6.10#23.56
```

**Comment by rsalvaterra** [comment]
> This does not build:

Ouch. Thanks for the heads up, @hauke. I use the tiny variant and it built just fine. Let me try to reproduce…

**Comment by rsalvaterra** [comment]
Ok, should be fixed now.

File: package/network/utils/iproute2/patches/001-fix-build.patch
Comment (robimarko) [inline package/network/utils/iproute2/patches/001-fix-build.patch:1]: Can you make this a full patch with description that is applicable by git am?

File: package/network/utils/iproute2/Makefile [hunk 1]
```diff
@@ -232,11 +225,6 @@ define Package/tc-full/install
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tc/tc $(1)/usr/libexec/tc-full
 endef
 
-define Package/tc-mod-iptables/install
-	$(INSTALL_DIR) $(1)/usr/lib/tc
-	$(CP) $(PKG_BUILD_DIR)/tc/m_xt.so $(1)/usr/lib/tc
-endef
```
Comment (hauke) [inline package/network/utils/iproute2/Makefile:238]: nice, the shared libs were removed here: https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/commit/?id=45dc10463dd5a51eef944b110a4f046feb1b46db

File: package/network/utils/iproute2/patches/001-fix-build.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Index: iproute2-6.8.0/bridge/mdb.c
+===================================================================
+--- iproute2-6.8.0.orig/bridge/mdb.c
++++ iproute2-6.8.0/bridge/mdb.c
+@@ -7,6 +7,7 @@
+ #include <stdlib.h>
+ #include <unistd.h>
+ #include <fcntl.h>
++#include <limits.h>
```
Comment (neheb) [inline package/network/utils/iproute2/patches/001-fix-build.patch:9]: uhhh, limit.h is included here since version 6.5.0

**Comment by neheb** [comment]
Please backport https://github.com/iproute2/iproute2/commit/64ed1886e719f40acb554ac895305acb43f27bea and https://github.com/iproute2/iproute2/commit/69e3b2fadcd32683db2942f31fe41f0fbb2185f8 instead.

**Comment by rsalvaterra** [comment]
Oh, I missed it. I guess I've been carrying this patch since forever. Will
fix.

A domingo, 12/05/2024, 21:38, Rosen Penev ***@***.***>
escreveu:

**Comment by neheb** [comment]
Also,

```
rdma.c: In function 'main':
rdma.c:110:20: error: implicit declaration of function 'basename'; did you mean 'rename'? [-Wimplicit-function-declaration]
  110 |         filename = basename(argv[0]);
      |                    ^~~~~~~~
      |                    rename
rdma.c:110:18: error: assignment to 'char *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  110 |         filename = basename(argv[0]);
      |                  ^
make[4]: *** [../config.mk:36: rdma.o] Error 1
make[4]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-i386_pentium-mmx_musl/linux-x86_legacy/iproute2-iptiny/iproute2-6.8.0/rdma'
make[3]: *** [Makefile:80: all] Error 2
make[3]: Leaving directory '/home/mangix/devstuff/openwrt/build_dir/target-i386_pentium-mmx_musl/linux-x86_legacy/iproute2-iptiny/iproute2-6.8.0'
make[2]: *** [Makefile:261: /home/mangix/devstuff/openwrt/build_dir/target-i386_pentium-mmx_musl/linux-x86_legacy/iproute2-iptiny/iproute2-6.8.0/.built] Error 2
make[2]: Leaving directory '/home/mangix/devstuff/openwrt/package/network/utils/iproute2'
time: package/network/utils/iproute2/iptiny/compile#17.27#4.60#16.21
    ERROR: package/network/utils/iproute2 failed to build (build variant: iptiny).
make[1]: *** [package/Makefile:129: package/network/utils/iproute2/compile] Error 1
make[1]: Leaving directory '/home/mangix/devstuff/openwrt'
make: *** [/home/mangix/devstuff/openwrt/include/toplevel.mk:233: package/iproute2/compile] Error 2
```

**Comment by neheb** [comment]
I'm noticing format errors relating to time_t and 32-bit.

**Comment by rsalvaterra** [comment]
Pull request updated, now at 6.9.0.

**Comment by neheb** [comment]
I'd prefer to rebase over https://github.com/openwrt/openwrt/pull/15480

**Comment by hauke** [comment]
I merged the iproute2 update in 2ae1330a22e62ddc9ac3e25daf974a2e01da11d3 using #15480.

**Pull request closed without merging**